src/js/utils/dateUtils.js

```diff
@@ -17,4 +27,149 @@ const dateWithoutTimeZone = ({
+  // Date object is automatically converted to the user's browser timezone, so the user doesn't have
+  // to set it on their own
+  return dateOnly
+    ? new Date(
+      parsedDate.getFullYear(),
+      parsedDate.getMonth(),
+      parsedDate.getDate(),
+    )
+    : parsedDate;
+};
+
+/**
+ * Converts a date to a string in specified format
+ * @param {Object} params
+ * @param {Object} options
+ * @param {Date} params.date - date object
+ * @param {String} params.dateFormat - output date format
+ * @param {String} options.locale - output locale
+ * @returns {String}
+ */
+export const formatDateToString = ({
+  date,
+  dateFormat = DateFormatDateFns.MMM_DD_YYYY,
+  options = {
+    locale: locales.enUS,
+  },
+}) => {
+  if (!date) {
+    return null;
+  }
+
+  return format(date, dateFormat, {
+    locale: options.locale,
+  });
+};
+
+/**
+ A method for converting Date to an ISO-formatted date-time string (for formatting API
+ request fields)
+ */
+export const formatDateToZonedDateTimeString = (date) => formatDateToString({
+  date,
+  dateFormat: DateFormatDateFns.MM_DD_YYYY_HH_MM_Z,
+});
+
+/**
+ * A method for converting Date to a localized date string for display (shifted time by timezone
+ * differences)
+ */
+export const formatDateToDatetimeString = (date, locale = locales.enUS) => formatDateToString({
+  date,
+  dateFormat: DateFormatDateFns.DD_MMM_YYYY,
+  options: {
+    locale,
+  },
+});
+
+/**
+ * A method for converting Date to a localized date string for display (skipping timezone)
+ */
+export const formatDateToDateOnlyString = (date, locale = locales.enUS) => {
+  const dateWithoutTimezone = new Date(
+    getYear(date),
+    getMonth(date),
+    getDate(date),
+  );
+
+  return formatDateToString({
+    date: dateWithoutTimezone,
+    dateFormat: DateFormatDateFns.DD_MMM_YYYY,
+    options: {
+      locale,
+    },
+  });
+};
+
+/**
+ * A method for formating ISO date string to date in another format
+ */
+export const formatISODate = (date, dateFormat) => format(parseISO(date), dateFormat);
+
+/**
+ * Get timezone offset, defaulting to the user's timezone offset
+ * @param {Number} timezoneOffset
+ * @returns {`${string}${string}:${string}`}
+ */
+export const displayTimezoneOffset = (timezoneOffset = new Date().getTimezoneOffset()) => {
+  // timezoneOffset = difference in minutes comparing to utc (timezoneOffset is an argument for
+  // testing purposes, because we can't force Date object to use different timezone that the user's)
+  const offsetMinutes = -timezoneOffset;
+  const sign = offsetMinutes >= 0 ? '+' : '-';
+
+  const absMinutes = Math.abs(offsetMinutes);
+  const hours = Math.floor(absMinutes / 60);
+  const minutes = absMinutes % 60;
+
+  // parsing hours / minutes to appropriate format: hours: 02, 01, 14, minutes: 02, 01, 59
+  const paddedHours = hours.toString().padStart(2, '0');
+  const paddedMinutes = minutes.toString().padStart(2, '0');
+
+  return `${sign}${paddedHours}:${paddedMinutes}`;
+};
```

Comment:
I kinda dont understand this method. What is it supposed to return and where it could be useful?